### PR TITLE
Add unit test for partitions with TRES=(null) (#864)

### LINKED ISSUE #864: auto_queues failed if TRES=(null) for partition
The parsing of TRES consider that the value cannot be null.
https://github.com/OSC/ood_core/blob/ea172947fb116a47a6b41a6807c8f67e7597f40e/lib/ood_core/job/adapters/slurm.rb#L377

We have clusters with dynamic partitions being configured which end into having TRES=(null)

`PartitionName=dynamic AllowGroups=ALL AllowAccounts=ALL AllowQos=ALL AllocNodes=ALL Default=NO QoS=N/A DefaultTime=NONE DisableRootJobs=NO ExclusiveUser=NO ExclusiveTopo=NO GraceTime=0 Hidden=NO MaxNodes=UNLIMITED MaxTime=UNLIMITED MinNodes=0 LLN=NO MaxCPUsPerNode=UNLIMITED MaxCPUsPerSocket=UNLIMITED NodeSets=dynamicns Nodes=(null) PriorityJobFactor=1 PriorityTier=1 RootOnly=NO ReqResv=NO OverSubscribe=NO OverTimeLimit=NONE PreemptMode=OFF State=UP TotalCPUs=0 TotalNodes=0 SelectTypeParameters=NONE JobDefaults=(null) DefMemPerNode=UNLIMITED MaxMemPerNode=UNLIMITED TRES=(null) ResumeTimeout=GLOBAL SuspendTimeout=GLOBAL SuspendTime=GLOBAL PowerDownOnIdle=NO`

```
App 18237 output: [2025-01-29 08:55:38 +0000 ]  WARN "Did not get queues from system with error wrong array length at 0 (expected 2, was 1)"
App 18237 output: [2025-01-29 08:55:38 +0000 ]  WARN "/opt/ood/gems/gems/ood_core-0.27.0/lib/ood_core/job/adapters/slurm.rb:377:in `to_h'\n/opt/ood/gems/gems/ood_core-0.27.0/lib/ood_core/job/adapters/slurm.rb:377:in `str_to_queue_info'\n/opt/ood/gems/gems/ood_core-0.27.0/lib/ood_core/job/adapters/slurm.rb:331:in `block (2 levels) in queues'\n/opt/ood/gems/gems/ood_core-0.27.0/lib/ood_core/job/adapters/slurm.rb:330:in `each_line'\n/opt/ood/gems/gems/ood_core-0.27.0/lib/ood_core/job/adapters/slurm.rb:330:in `block in queues'\n<internal:kernel>:90:in `tap'\n/opt/ood/gems/gems/ood_core-0.27.0/lib/ood_core/job/adapters/slurm.rb:329:in `queues'\n/opt/ood/gems/gems/ood_core-0.27.0/lib/ood_core/job/adapters/slurm.rb:696:in `queues'\n/var/www/ood/apps/sys/dashboard/app/lib/account_cache.rb:152:in `block (3 levels) in queues_per_cluster'\n/opt/ood/gems/gems/ood_core-0.27.0/lib/ood_core/clu
```

### PR DESCRIPTION
**Pull request by adhati09**
## What does this PR do?
Adds a unit test and fixture to verify that the Slurm adapter correctly parses partitions when TRES=(null). This ensures adapter.queues handles unconfigured or null TRES values gracefully without throwing exceptions.

## Related issue
Fixes #864 

## Testing
- [X] Tests included
- [ ] No tests needed — reason: ___

## Checklist
- [X] Follows project code style and conventions
- [ ] Documentation provided *(if new feature, adapter or behavior change)*
- [ ] This is a large feature and was discussed in an issue first *(if applicable)*

## Anything else?
     Added:

spec/fixtures/output/slurm/scontrol_show_part_tres_null.txt fixture containing output with TRES=(null).

test_queues_with_tres_null test method in test/job/adapters/slurm_test.rb.

### CODE REVIEW

File: spec/fixtures/output/slurm/scontrol_show_part_tres_null.txt
Comment (Bubballoo3) [inline spec/fixtures/output/slurm/scontrol_show_part_tres_null.txt:1]: Rather than start a new file for this line, let's just add it to the `owens_partitions.txt` fixture. There will be one rspec test that reads a list of partitions from that file and will need to be updated but I think that is preferable

File: test/job/adapters/slurm_test.rb [hunk 1]
```diff
@@ -103,4 +103,18 @@ def test_nil_parse_time
...
+         .returns([File.read('spec/fixtures/output/slurm/scontrol_show_part_tres_null.txt'), '', exit_success])
+
+    queues = adapter.queues
+    assert_equal(1, queues.size)
+
+    queue = queues.first
+    assert_equal('dynamic', queue.name)
+    assert_equal({}, queue.tres)
+  end
+
```
Comment (Bubballoo3) [inline test/job/adapters/slurm_test.rb]: A minor point, but this blank line between the method end and the class end should be removed

**Review by Bubballoo3 — DISMISSED** [review]
I think the test content looks good, just a few minor points on fixture placement and style

**Review by johrstrom — APPROVED** [review]
LGTM, thanks to fixing this and thanks @Bubballoo3 for the review.
